samples/algorithms/order-finding/Program.qs

```diff
@@ -0,0 +1,139 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+namespace Microsoft.Quantum.Samples.OrderFinding {
+    open Microsoft.Quantum.Intrinsic;
+    open Microsoft.Quantum.Arrays;
+    open Microsoft.Quantum.Convert;
+    open Microsoft.Quantum.Math;
+
+    @EntryPoint()
+    operation GuessOrder(index : Int) : Unit {
+        let perm = [1, 2, 3, 0];
+        let shots = 1024;
+
+        // print some info on the permutation
+        Message($"Permutation: {perm}");
+        Message($"Find cycle length at index {index}\n");
+
+        // early exit for index out of bounds 
+        if (index >= Length(perm)){
+            Message("Index can only be between 0 and 3 inclusive");
+            return ();
+        }
+
+        // compute exact order
+        Message($"Exact Order: {ComputeOrder(index, perm)}");
+
+        // guess order classically
+        Message("\nGuess classically:");
+        GuessOrderClassical(index, perm, shots);
+
+        // guess order quantum computationally
+        Message("\nGuess Quantum Computationally");
+        GuessOrderQuantum(index, perm, shots);
+    }
+    
+    /// # Summary
+    /// Returns the exact order (length) of the cycle that contains a given index.
+    function ComputeOrder(index : Int, perm : Int[]) : Int {
+        // ...
+        mutable order = 1;
+        mutable cur = index;
+
+        while (index != perm[cur]) {
+            set order = order + 1;
+            set cur = perm[cur];
+        }
+
+        return order;
+    }
+
+    /// # Summary
+    /// Guesses the order classically by applying estimate `shots` many times and returning the percentage for each order that was returned.
+    operation GuessOrderClassical(index : Int, perm : Int[], shots : Int) : Unit {
```

Comment:
`GuessOrderClassical` and `GuessOrderQuantum` are very similar and might be unified into a single operation which takes as input an operation.

Reply:
Thanks for this suggestion. I have combined them into a single operation taking the corresponding operation as input.